providers/implementations/signature/rsa.c

```diff
@@ -86,11 +86,7 @@ typedef struct {
      * by their Final function.
      */
     unsigned int flag_allow_md : 1;
-
-    /* The Algorithm Identifier of the combined signature algorithm */
-    unsigned char aid_buf[128];
-    unsigned char *aid;
-    size_t  aid_len;
+    unsigned int mgf1_md_set : 1;
```

Comment:
caching the AID is bad because...?

Reply:
Computing it when parameters are set does not make sense, because the parameters can be set in multiple passes. We could cache the value when we compute it in the get_params, but as usually the AID is queried with get_params just once per signature operation, I do not think it is worth caching it. There would also have to be a cache invalidation logic and so on. I do not really think it is worth the code complication also because computing the AID is quite cheap operation.

Reply:
Okie, good enough for me